docs/src/main/tut/docs/config-cursors.md

```diff
@@ -0,0 +1,89 @@
+---
+layout: docs
+title: Config Cursors
+---
+
+## {{page.title}}
+
+When a `ConfigReader` needs to be created from scratch, users need to implement a `from` method with the following
+signature:
+
+```scala
+def from(cur: ConfigCursor): Either[ConfigReaderFailures, A]
+```
+
+The `ConfigCursor` class is a wrapper for the raw `ConfigValue` provided by Typesafe Config. It provides an idiomatic,
+typesafe API for the most common operations needed while reading a config. In particular, it provides cast operations
+and key accesses that integrate neatly with the [PureConfig errors API](error-handling.html). When using cursors
+properly, most errors are automatically handled and filled with rich information about the location of the failure.
+
+We'll show how to implement our own `ConfigReader` for the following class:
+
+```tut:silent
+class Person(firstName: String, lastName: String) {
+  override def toString = s"Person($firstName $lastName)"
+}
+
+case class Conf(person: Person)
+```
+
+We intend our config to look like this:
+
+```tut:silent
+import com.typesafe.config.ConfigFactory
+
+val conf = ConfigFactory.parseString("person = { name: John Doe }")
+```
+
+For the purposes of this example, we'll assume the provided `name` will always have at least two words.
+
+An implementation of the `ConfigReader` using the cursors API is shown below:
+
+```tut:silent
+import pureconfig._
+
+def firstNameOf(name: String): String =
+  name.substring(0, name.indexOf(' '))
```

Comment:
It seems a bit of a shame that these methods don't report errors nicely if the `name` doesn't include a string. It seems like including that functionality in this example would be useful to anyone else attempting to write a pure config reader that doesn't throw.

Reply:
Yes, I agree with you. I tried to come up with a non-trivial example that didn't involve using the contents of the "Error Handling" section (i.e. having to create failure reasons) here, but I couldn't find a better one. I wanted to focus this example on the automatic error handling and not on the explicit handling of application-level errors.

At 2e5b19a I modified the example so that at least it doesn't throw in any case and the new domain (capturing all names instead of the last one) does not require as much validations. If you think of any other suggestion please let me know :)